web/src/main/webapp/loadMenu.ts

```diff
@@ -238,12 +238,8 @@ export class LoadMenu extends RemoteObject implements IDataView {
                 text: "Federated DB tables...",
                 action: () => {
                     const dialog = new DBDialog(true);
-                    const connection = dialog.getCassandraConnection();
-                    if (connection == null) {
-                        this.page.reportError("Not a valid connection");
-                        return;
-                    }
-                    dialog.setAction(() => this.init.loadFederatedDBTable(dialog.getConnection(), connection, this.page));
+                    dialog.setAction(() => this.init.loadFederatedDBTable(dialog.getJdbcConnection(),
```

Comment:
how about instead checking if this is cassandra and then extracting only the proper connection?
Probably loadFederatedDBTable should have a single argument, not two.

Reply:
I updated this code. The connection object will be passed as an `any` object. Thus, it can handle multiple connection types including HiveConnection without adding more input argument.